sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java

```diff
@@ -322,6 +323,13 @@ public String create(PipelineOptions options) {
   }
 
   /**
+   * Return a map of properties which correspond to {@link RuntimeValueProvider},
+   * keyed by the property name.  The value is a map containing type and default
+   * information.
+   */
+  Map<String, Map<String, Object>> outputRuntimeOptions();
```

Comment:
Who's this method intended for? My first guess it should be public.

Assuming it is public (not applicable otherwise):
* Naming: outputRuntimeOptions vs. getRuntimeOptions vs. mutableRuntimeOptions vs. other alternatives. Not sure yet.
* Return value: `Map<String, Map<String, Object>>` is a bit sketchy and hard to modify/extend in the future. Any other alternatives?